Tests/configure_and_test_integration_instances.py

```diff
@@ -0,0 +1,406 @@
+    '''Set integration module values for integration instance creation
+
+    The integration_configuration and integration_params should match, in that
+    they are for the same integration
+
+    Arguments:
+        integration_configuration: (dict)
+            dictionary of the integration configuration parameters/keys that need
+            filling to instantiate an instance of a given integration
+        integration_params: (dict)
+            values for a given integration taken from the configuration file in
+            which the secret values are stored to configure instances of various
+            integrations
+        integration_instance_name: (str)
+            The name of the integration instance being configured if there is one
+            provided in the conf.json
+        is_byoi: (bool)
+            If the integration is byoi or not
+
+    Returns:
+        (dict): The configured module instance to send to the Demisto server for
+        instantiation.
+    '''
+    module_configuration = integration_configuration.get('configuration', {})
+    if not module_configuration:
+        module_configuration = []
+
+    instance_name = '{}_test_{}'.format(integration_instance_name.replace(' ', '_'),
+                                        str(uuid.uuid4()))
+    # define module instance
+    module_instance = {
+        'brand': integration_configuration['name'],
+        'category': integration_configuration['category'],
+        'configuration': integration_configuration,
+        'data': [],
+        'enabled': "true",
+        'engine': '',
+        'id': '',
+        'isIntegrationScript': is_byoi,
+        'name': instance_name,
+        'passwordProtected': False,
+        'version': 0
+    }
+
+    # set module params
+    for param_conf in module_configuration:
+        if param_conf['display'] in integration_params or param_conf['name'] in integration_params:
+            # param defined in conf
+            key = param_conf['display'] if param_conf['display'] in integration_params else param_conf['name']
+            if key == 'credentials':
+                credentials = integration_params[key]
+                param_value = {
+                    'credential': '',
+                    'identifier': credentials['identifier'],
+                    'password': credentials['password'],
+                    'passwordChanged': False
+                }
+            else:
+                param_value = integration_params[key]
+
+            param_conf['value'] = param_value
+            param_conf['hasvalue'] = True
+        elif param_conf['defaultValue']:
+            # if the parameter doesn't have a value provided in the integration's configuration values
+            # but does have a default value then assign it to the parameter for the module instance
+            param_conf['value'] = param_conf['defaultValue']
+        module_instance['data'].append(param_conf)
+
+    return module_instance
+
+
+def main():
```

Comment:
this function is like reading a book, split this too several more readable functions please

Reply:
please, don't make me suffer reading it